tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+            // the packet is not the oldest sentVSCPacket, something went wrong
+            Err("Received maturation is not for the oldest sentVSCPacket")
+        } else {
+            val currentReceivedMaturations = currentState.providerState.receivedMaturations
+            val newReceivedMaturations = currentReceivedMaturations.union(Set(packet))
+            val newProviderState = currentState.providerState.with(
+                "receivedMaturations", newReceivedMaturations
+            )
+            // prune the sentVSCPacket
+            val newSentVSCPacket = currentState.providerState.sentVSCPackets.get(sender).tail()
+            val newState = currentState.with(
+                "providerState", newProviderState
+            )
+            Ok(newState)
+        }
+    }
+
+    // removes the oldest outstanding packet from the consumer. on-chain, this would happen when the packet is acknowledged.
+    // only the oldest packet can be removed, since we model ordered channels.
+    pure def removeOutstandingPacketFromConsumer(currentState: ProtocolState, sender: Chain): Result = {
+        val currentOutstandingPackets = currentState.consumerStates.get(sender).outstandingPacketsToProvider
+        val newOutstandingPackets = currentOutstandingPackets.tail()
+        val newConsumerState = currentState.consumerStates.get(sender).with(
+            "outstandingPacketsToProvider", newOutstandingPackets
+        )
+        val newConsumerStates = currentState.consumerStates.set(sender, newConsumerState)
+        val newState = currentState.with(
+            "consumerStates", newConsumerStates
+        )
+        Ok(newState)
+    }
+
+    // removes the oldest outstanding packet (to the given consumer) from the provider.
+    // on-chain, this would happen when the packet is acknowledged.
+    // only the oldest packet can be removed, since we model ordered channels.
+    pure def removeOutstandingPacketFromProvider(currentState: ProtocolState, receiver: Chain): Result = {
+        val currentOutstandingPackets = currentState.providerState.outstandingPacketsToConsumer.get(receiver)
+        val newOutstandingPackets = currentOutstandingPackets.tail()
+        val newProviderState = currentState.providerState.with(
+            "outstandingPacketsToConsumer", 
+            currentState.providerState.outstandingPacketsToConsumer.set(receiver, newOutstandingPackets)
+        )
+        val newState = currentState.with(
+            "providerState", newProviderState
+        )
+        Ok(newState)
+    }
+
+    // Updates the given oldValidatorSet by setting the validator to newVotingPower.
+    // If newVotingPower is zero, the validator is removed.
+    pure def getUpdatedValidatorSet(oldValidatorSet: ValidatorSet, validator: Node, newVotingPower: int): ValidatorSet =
+        if (newVotingPower > 0) 
+        oldValidatorSet.put(validator, newVotingPower)
+        else
+        oldValidatorSet.mapRemove(validator)
```

Comment:
Similar to a [previous comment](https://github.com/cosmos/interchain-security/pull/1336/files#r1348633598). It feels we might unnecessarily complicate our lives by removing validators and we shouldn't consider this case?

Reply:
I think setting to 0 should be allowed, since its the way we would expect validators to get removed from the set.
Whether removing or setting to 0 is better is the right question imo

Reply:
I don't have strong feelings on this.

>Whether removing from the set or treating 0 are removed is better is the right question imo

Good point. My take is that this currently tests that a validator gets removed when voting power is 0 but this does  not seem to be part of the CCV protocol. It seems to me that it is testing more  the `staking` module instead.
We mainly care that `VSCPacket`s are correctly propagated to the consumer chain -- why would it make a difference if the validator shows up with voting power 0 or actually removed?

Additionally, if we believe that removing nodes tests something more, would it make sense to be able to add nodes as well?

Reply:
Ok, that convinces me, this is not really part of CCV. I will just allow 0 by setting the validator to 0 for simplicity. We can imagine it means the val is unbonded, but no special behaviour